### DISCUSSION TITLE
How do you setup CI now that Apple forces new accounts to have 2fa?

### DISCUSSION THREAD

**Discussion by rodrigoelp** [fastlane/fastlane#18097] [Q&A]
I've been trying to understand how to setup fastlane now that Apple enforces the usage of 2fa in a process that is not controller by fastlane.
We have an old account in which we haven't set 2fa and that has been working fine but, we received a notification we had to accept privacy terms today and we thought to split some of our builds into multiple accounts.
I have been through some of the documentation and it basically tells you "do not use 2fa" (which you can't turn off now)... so, how do you guys set up a ci that is meant to run builds even in the middle of the night?

**Comment by rodrigoelp** [comment]
Just to provide a little more context, our build configuration produces a development ipa (which we only use for smoke testing) and this one can be configured to use an appstore connect api key (https://docs.fastlane.tools/app-store-connect-api/) and it works fine... But there is no api key we can use for an enterprise signed api (which we also produce for a wider local qa crowd).

**Comment by NikKovIos** [comment]
I'm trying to use `app_store_connect_api_key `, but have an issue:
https://github.com/fastlane/fastlane/discussions/18100

**Reply by rodrigoelp** [reply]
A just read your issue, I got around that issue by providing an empty username

**Reply by NikKovIos** [reply]
Username where?

**Reply by rodrigoelp** [reply]
In `pilot`, I noticed with `sigh` you still need to provide the `username` even when you use the api key. Basically our script is:

```ruby
# sample.
lane :build_something do |options|
    app_store_connect_api_key(
        key_id: "MZ5***8KG4",
        issuer_id: "69a6de90-****-****-****-5b8c7c11a4d1",
        key_filepath: "./fastlane/AuthKey_API_MZ5***8KG4.p8",
        duration: 7200, # optional
        in_house: false, # optional but may be required if using match/sigh
      )
    sigh(
        app_identifier: "org.sample.app",
        team_id: "XXX",
        username: "this is unused", # you can literally provide any value here, it doesn't matter what but it can't be empty or nil.
        provisioning_name: "Super Dupper App.",
        readonly: true
    )
end
```

Maybe this is an issue particular of `sigh`. I know the documentation has the sample with `pilot` but we couldn't get it to work that way and `app_store_connect_api_key` sets the lane_context either way that other fastlane functions call.

**Comment by j3ski-passion** [comment]
We have set up a cron job on our server to log in periodically & an sms forwarding mechanism. Then forwarding the cookie to CI. Will see how it goes...

**Comment by ffittschen** [comment]
I summarized some learnings we had about using the `FASTLANE_SESSION` workaround for an Enterprise account and also added some tips for a 2FA setup that is decoupled from using just 1 phone number: https://github.com/fastlane/fastlane/discussions/18113
